diffrax/_brownian/tree.py

```diff
@@ -83,19 +84,39 @@ def _levy_diff(x0: LevyVal, x1: LevyVal) -> LevyVal:
     `LevyVal(W_su, H_su)`
     """
 
-    su = jnp.asarray(x1.dt - x0.dt, dtype=x0.W.dtype)
-    w_su = x1.W - x0.W
-    if x0.H is None or x1.H is None:  # BM only case
-        return LevyVal(dt=su, W=w_su, H=None, bar_H=None, K=None, bar_K=None)
-
-    assert (x0.bar_H is not None) and (x1.bar_H is not None)
-    # if we are at this point levy_area == "space-time"
-    _su = jnp.where(jnp.abs(su) < jnp.finfo(su).eps, jnp.inf, su)
-    inverse_su = 1 / _su
-    u_bb_s = x1.dt * x0.W - x0.dt * x1.W
-    bhh_su = x1.bar_H - x0.bar_H - 0.5 * u_bb_s  # bhh_su = H_{s,u} * (u-s)
-    hh_su = inverse_su * bhh_su
-    return LevyVal(dt=su, W=w_su, H=hh_su, bar_H=None, K=None, bar_K=None)
+    if len(x0) == 2:  # BM only case
+        assert len(x1) == 2
+        dt0, w0 = x0
+        dt1, w1 = x1
+        su = jnp.asarray(dt1 - dt0, dtype=w0.dtype)
+        return LevyVal(dt=su, W=w1 - w0, H=None, K=None)
+
+    elif len(x0) == 4:  # space-time levy area case
```

Comment:
I don't feel so strongly that I think this needs to be changed now, but if I was to do this again then I would recommend creating an internal-only type that's used across this interface, so that we can use safer `if isinstance(x0, Foo): x0.bar` checks and lookups.

(Maybe let's do that whenever space-time-time gets added here, to help avoid any bugs then.)